modules/nextflow/src/main/groovy/nextflow/Session.groovy

```diff
@@ -1223,10 +1223,10 @@ class Session implements ISession {
             log.trace "Clean workdir complete"
         }
         catch( Exception e ) {
-            log.warn("Failed to cleanup work dir: ${workDir.toUriString()}")
+            log.warn("Failed to cleanup work dir: ${workDir.toUriString()}. Exception: ${e.message}")
         }
         finally {
-            db.close()
+            db?.close()
```

Comment:
This can be better managed using (new) try-with-resource statement ie. 

```
try ( CacheDB db = CacheFactory.create(uniqueId, runName).openForRead() ) {
  ..
}
catch( Exception e ) {
  .. 
}

Reply:
Ah, I didn't know about try-with-resource! Lovely feature, no implemented in the latest commit.